includes/class-newspack-popups-segmentation.php

```diff
@@ -181,5 +184,36 @@ function( $cat ) {
 	public static function get_segmentation_endpoint() {
 		return plugins_url( '../api/segmentation/index.php', __FILE__ );
 	}
+
+	/**
+	 * Create the clients and events tables.
+	 */
+	public static function create_database_table() {
+		global $wpdb;
+		$events_table_name = Segmentation::get_events_table_name();
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching
+		if ( $wpdb->get_var( $wpdb->prepare( 'SHOW TABLES LIKE %s', $events_table_name ) ) != $events_table_name ) {
+			$charset_collate = $wpdb->get_charset_collate();
+
+			$sql = "CREATE TABLE $events_table_name (
+				id bigint(20) NOT NULL AUTO_INCREMENT,
+				created_at date NOT NULL,
+				-- type of event
+				type varchar(20) NOT NULL,
```

Comment:
It would make sense to add a default of `post_read` for the `type` column to avoid adding that to all `INSERT`s.

Reply:
True, but in the future there will be other event types, and this would introduce a lot of forking in `parse_events_logs` and two separate calls to `bulk_db_insert` method – one for read posts (omitting the event type column) and one for other events. I don't think adding that complexity is worth it.